### ISSUE TITLE
Incorrect "Add 'this.'" diagnostic for collection initializers

### ISSUE THREAD

**Issue by stijnherreman** [dotnet/roslyn#30288]
**Version Used**: 15.8.6

**Steps to Reproduce**:

Create a new .NET Standard Class Library project.

Add the following `.editorconfig` file:

```
root = true

[*.cs]
dotnet_style_qualification_for_method = true:error
```

Add the following `Class1.cs` file:

```c#
using System.Collections.Generic;

namespace ClassLibrary1
{
    public class Foo
    {
        public string Bar { get; set; }
    }

    public class Baz
    {
        public Baz()
        {
            var foo = new Foo() { Bar = "bar" };

            var foos = new List<Foo>()
            {
                new Foo() { Bar = "bar" }
            };
        }
    }
}
```

**Expected Behavior**: No IDE0009 error.

**Actual Behavior**: An IDE0009 error is listed for the `new Foo() { Bar = "bar" }` line in the collection initializer.

![image](https://user-images.githubusercontent.com/1618054/46420555-9c556580-c730-11e8-939e-7a95ce4fac6e.png)

Note: this is related to #22776, which is about the same problem for class initializers.

**Comment by stijnherreman** [comment]
Duplicate of #29413.

**Comment by sharwell** [comment]
Duplicate of #28509
